ktor-server/ktor-server-plugins/ktor-server-websockets/jvmAndNix/src/io/ktor/server/websocket/WebSocketUpgrade.kt

```diff
@@ -96,8 +96,11 @@ public class WebSocketUpgrade(
 
         webSocket.launch(WebSocketHandlerCoroutineName) {
             try {
-                webSocket.start(handle)
+                webSocket.handle()
+                webSocket.flush()
             } catch (cause: Throwable) {
+            } finally {
+                webSocket.cancel()
```

Comment:
could you explain why we need cancel here?

Reply:
it's inlined `start` function:
https://github.com/ktorio/ktor/pull/2899/files/7ec0dc58a9ecdf887347ee55b037d3e04cc7817a#diff-b3700a64a6cb2f618bcaf420986cddf505e8fc5de9aff09b175b3f6bc3cd0ff6L85-L92

Current behavior is that after `handler` lambda is finished, we should close websocket, so call `cancel`. I didn't want to change behavior here.